src/start.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { getEnvBoolean } from './utils';
+import { startMetrics, MetricsOptions } from './metrics';
+import { startProfiling, ProfilingOptions } from './profiling';
+import { startTracing, TracingOptions } from './tracing';
+
+interface GeneralOptions {
+  accessToken: string;
```

Comment:
By reading the types, `accessToken` seems to be required, even though `Partial<Options>` is used in `start`. Are there any downsides to making it optional here?

Reply:
Are there any upsides to making all the properties optional? The type for `start` is still `Partial<GeneralOptions>` and GeneralOptions is used nowhere for now.

Reply:
The only upside I thought about is the readability/glance value when going over the types. And I might have missed it, but is `GeneralOptions` used anywhere else? Could merge the 2 fields into `Options` itself

Reply:
Sure! I had it separate since I anticipated needing to type only general options somewhere. Merging everything into one for now.